source/Connection.cpp

```diff
@@ -212,10 +217,11 @@ Connection::GetConnectionOutput(OutChanProps outChan)
 }
 
 void
-Connection::setEnv(EnvCond* env_in, moordyn::Waves* waves_in)
+Connection::setEnv(EnvCond* env_in, moordyn::Waves* waves_in, std::shared_ptr<moordyn::Seafloor> seafloor_in)
```

Comment:
It is a bit strange that we are passing by pointers and just seafloor is using std::shared_ptr. Probably we shall consider moving everything to shared_ptrs

Reply:
Yeah, I also think it would make sense to move to shared_ptrs. I will discuss with Alex (our new guy) and see if he sees any reason not to - I think the raw pointers are just a holdover from the initial C->C++ conversion. 

Really, what we need is to spend a week or two just refactoring things like this.

Reply:
@AlexWKinley addressed this in ddd95a8144288ce8e084aa28037ae6ffd561f842